infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -260,6 +309,28 @@ public LogReplicationEntryMsg resend() {
         return ack;
     }
 
+    private void activateBackpressure() {
+        if (!isBackpressureActive) {
+            log.info("Log entry ACKs timing out on sink, allocating additional time for retries.");
+            isBackpressureActive = true;
```

Comment:
Why do you need to switch this on/off?

Reply:
I saw some getters/setters being used for the timeoutResponse within the failure detector, shouldn't be applicable for LR's router implementation but to keep the value consistent when we want to add a retry the on/off just updates the static field and resets to the original value it has cached, also related to your above comment. But if there's no need for that field to be real time accurate, overloading `sendRequestAndGetCompletable` is a cleaner approach